src/ParallelHashMap.h

```diff
@@ -566,23 +553,47 @@ void ParallelHashMap<K,V>::insert(K key, V value)
     return;
 
   /* get lock hash */
-  #ifdef OPENMP
   size_t lock_hash = (std::hash<K>()(key) & (_table->bucket_count() - 1))
     % _num_locks;
 
   /* acquire lock */
   omp_set_lock(&_locks[lock_hash]);
-  #endif
 
   /* insert value */
   _table->insert(key, value);
 
   /* release lock */
-  #ifdef OPENMP
   omp_unset_lock(&_locks[lock_hash]);
-  #endif
 }
 
+
+/**
+ * @brief Updates the value associated with a key in the parallel hash map.
+ * @details The thread first acquires the lock for the bucket associated with
+ *      the key is acquired, then the linked list in the bucket is searched
+ *      for the key. If the key is not found, an exception is returned. When
+ *      the key is found, the value is updated and the lock is released.
+ * @param key the key of the key/value pair to be updated
+ * @param value the new value for the key/value pair
+ */
+template <class K, class V>
+void ParallelHashMap<K,V>::update(K key, V value) {
```

Comment:
@geogunow added this. Maybe I hadn't merged with his latest PR or something. I'll merge with develop and see if that changes anything.